docs: Add user guide for emoji

1Niels opened the pull request:
Hey, I added new page with a guide for using emoji in messages and removed the section about emoji from the index. Furthermore I fixed the links for:

- Create a stream
- Streams and private messages
- signing out
- message formatting

Thanks!

smarx commented:
*Automated message from Dropbox CLA bot*

@1Niels, it looks like you've already signed the Dropbox CLA. Thanks!

arpith commented:
Hey @1Niels :) is there a particular reason for calling it Emoji Code? I think the earlier guide called it emoji name. A few examples here would help, as well as explaining that the pop-up menu shows the first five emojis whose names contain the letters typed. (I'm sure you have a better way of explaining this than me :-)

1Niels commented:
@arpith I called them Emoji code because that's what they're called on [Slack's emoji guide](https://get.slack.help/hc/en-us/articles/202931348-Emoji-and-emoticons) and more commonly only other websites as well. I think I will probably change the section name from **Emoji Code** to **Using emoji codes** and I'll include your suggestion in the last step. Thanks for the feedback!

1Niels commented:
Hey, I changed the section name and moved the note at the bottom of the page under the second step of **Using emoji codes**. You can see on my dev server [here](http://1niels.zulipdev.org:9991/help/emoji). Thanks!

timabbott commented on templates/zerver/help/emoji.md:
```diff
@@ -0,0 +1,27 @@
+
+1. Copy any emoji you would like to include onto your clipboard.
+2. Create either a "**New stream message**" or a "**New private message**".
+3. Past the emoji into the message box.
```
Might be good to give an example here.

timabbott commented on templates/zerver/help/emoji.md:
```diff
@@ -0,0 +1,27 @@
+Emoji allow you to add a little character and some fun to your messages! This guide shows you how to send emoji by typing the emoji code, pasting Unicode, or using Zulip’s built-in emoji menu.
+
+### Using the emoji menu
+1. Create either a "**New stream message**" or a "**New private message**".
```
For this, I'd probably just stay "start composing or editing a message, e.g. by clicking on a message to start a reply".  My reasoning here is that usually the way people start composing messages is by clicking on them to reply.

This feedback also applies to the below text.

TomaszKolek commented:
@1Niels Could you please rebase your branch with actual master?

1Niels commented:
@TomaszKolek Yes, I'll work on that now.

mjec commented:
This is looking good @1Niels but unfortunatley needs to be rebased against master again. As soon as that's done I think it will be good to go!

mjec commented:
Thanks @1Niels this looks great!

@timabbott I think this is ready to merge

timabbott commented:
Merged, thanks @1Niels!!!

1Niels commented:
@mjec Thanks!
@timabbott No problem!
